docs/docs/code/python/delay/README.md

```diff
@@ -0,0 +1,124 @@
+---
+short_description: How to delay a workflow's execution with Python.
+thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646841376/docs/icons/icons8-time-96_kupxpi.png
+---
+
+# Delaying a workflow
+
+Use `pd.flow.delay` to [delay a step in a workflow](/workflows/flow-control/#delay).
+
+These docs show you how to write Python code to handle delays. If you don't need to write code, see [our built-in delay actions](/workflows/flow-control/#delay-actions).
+
+[[toc]]
+
+## Using `pd.flow.delay`
+
+`pd.flow.delay` takes one argument: the number of **milliseconds** you'd like to pause your workflow until the next step executes. {{$site.themeConfig.DELAY_MIN_MAX_TIME}}.
+
+Note that [delays happen at the end of the step where they're called](#when-delays-happen).
+
+```python
+import random
+
+def handler(pd: 'pipedream'):
+    # Delay a workflow for 60 seconds (60,000 ms)
+    pd.flow.delay(60 * 1000)
+
+    # Delay a workflow for 15 minutes
+    pd.flow.delay(15 * 60 * 1000)
+
+    # Delay a workflow based on the value of incoming event data,
+    # or default to 60 seconds if that variable is undefined
+    default = 60 * 1000
+    delayMs = pd.steps["trigger"].get("event", {}).get("body", {}).get("delayMs", default)
+    pd.flow.delay(delayMs)
+
+    # Delay a workflow a random amount of time
+    pd.flow.delay(random.randint(0, 999))
+```
+
+::: tip Paused workflow state
+
+When `pd.flow.delay` is executed in a Python step, the workflow itself will enter a **Paused** state.
+
+While the workflow is paused, it will not incur any credits towards compute time. You can also [view all paused workflows in the Event History](/event-history/#filtering-by-status).
+
+:::
+
+### Credit usage
+
+The length of time a workflow is delayed from `pd.flow.delay` does _not_ impact your credit usage. For example, delaying a 256 megabyte workflow for five minutes will **not** incur ten credits.
+
+However, using `pd.flow.delay` in a workflow will incur two credits.
```

Comment:
Do you want to clarify that it will use at least 2 credits? It could be more, depending on the duration and memory configuration.

Reply:
That makes sense. I added a `::: tip` about duration and memory affecting credit usage in https://github.com/PipedreamHQ/pipedream/pull/10192/commits/38e621a2583b8db148562629afaade91b482680e.
What do you think?
https://pipedream-docs-git-python-docs-improvements-312-pipedreamers.vercel.app/docs/code/python/delay/#credit-usage